### ISSUE TITLE
docparams check doesn't understand class hierarchies

### ISSUE THREAD

**Issue by kasium** [PyCQA/pylint#4955]
### Bug description

```python
class MyError(ValueError):
    pass

def foo():
    """Foo method

    Raises:
        ValueError: all the time
    """
    raise MyError()
```

### Command used

```shell
pylint x.py --disable all --load-plugins pylint.extensions.docparams --enable missing-raises-doc
```

### Pylint output

```shell
************* Module x
x.py:5:0: W9006: "MyError" not documented as being raised (missing-raises-doc)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 2.50/10, +5.00)
```

### Expected behavior

No error;

I understand that checking the hierarchy is more complicated, but I guess there are valid use cases to expose only a general exception

### Pylint version

```shell
pylint: 2.10.2
astroid: 2.7.3
python: 3.7.1
```

### OS / Environment

Linux

**Comment by kasium** [comment]
@Pierre-Sassoulas I would like to give it a try.

My current idea is to enhance pylint/extensions/_check_docs_utils.py::possible_exc_types to not only return the names of the exceptions, but the whole exception hierarchy. Or, as an alternative, it can return a set of nodes instead of strings to check later the hierarchy.

But, I currently struggle to get the super class of a given exception class. Any idea?

**Comment by kasium** [comment]
@Pierre-Sassoulas thanks a lot. Found it.

By any chance, do you know why `astroid.Instance` is checked here? https://github.com/PyCQA/pylint/blob/6a7209b9b8b9bbd0fdf08ce22daf325c9d0d28ec/pylint/extensions/_check_docs_utils.py#L162

I already removed the type and ran `pytest tests/extensions/test_check_raise_docs.py ` which didn't showed any failing test. No clue how to handle that type

**Comment by Pierre-Sassoulas** [comment]
I don't know, you could try to check the git history of this particular line. See https://stackoverflow.com/a/19757493/2519059

**Comment by kasium** [comment]
Already done. I found the commit adding it but there was no explanation why this is needed. Let me further check
